build.sbt

```diff
@@ -16,7 +16,7 @@ lazy val commonSettings = Seq(
   organization := "edu.berkeley.cs",
   version := "3.6-SNAPSHOT",
   autoAPIMappings := true,
-  scalaVersion := "2.12.17",
+  scalaVersion := "2.13.10",
```

Comment:
We should do this but not in a warning cleanup PR, switching the default version to 2.13 is not a huge deal but worth calling out. Anyone using Chisel as a submodule and using the SBT build will have a bad time (how rocket-chip used to use Chisel), and it is worth mentioning that this will affect developers because they will do things or use APIs that will then fail upon 2.12 compilation (vice versa rarely happens, but does sometimes too).

Reply:
Created separate PR #2978 to bump the version